In [ ]:
#1 
A> Internship
b> 2.5


In [ ]:
#2
 answer- 1

In [ ]:
#3 

170.54+0.15x ( solved the equation in pen and paper)

In [ ]:
#4

T= unstandardized Coifficient of Crime/standard error of Crime

-9.460=coifficient of crime/0.044
coifficient of crime= -0.41624

for every unit increase in Crime Rate, the property price will decrease at the rate of 0.41624

In [ ]:
#5
 from the data estimated value of parameter = 9.46*0.044= 0.41624
    Given parameter =0.3
    
    t- statastics = 0.41624-0.3/0.044 = 2.6418 (unit)
    
    t value at alpha 0.05 is 1.647
    
    the calculated statastics is greater than t critical value
    implies its a right tail test
    and Therefore, it can be concluded that the claim of Francis Galton that for every unit increase in crime rate, 
    the price will decrease by at least INR 30,000 is valid.

In [58]:
#6
# Import required libraries
import numpy as np                  #data handling
import pandas as pd  

import math
import scipy.stats as st            #Statistics

import matplotlib.pyplot as plt     #Visualization
import seaborn as sns 
import statsmodels.api as sm

In [59]:
#6
truck = pd.read_excel("MLR_TruckMaintenance.xlsx", sheet_name='Data')
truck.head()

Truck  Maintenance Expense  Miles Driven  Age of Truck
0      1               908.56         10500            10
1      2               751.12          9700             7
2      3               793.55          9200             8
3      4               619.61          8300             9
4      5               380.11          6500             5

In [26]:
truck.columns


Index(['Truck', 'Maintenance Expense', 'Miles Driven', 'Age of Truck'], dtype='object')

In [60]:
# Training a linear regression model for maintenance Expense ~ X  
X=truck.drop(columns='Maintenance Expense')
Y = truck['Maintenance Expense']

X = truck.drop(['Maintenance Expense','Truck'], axis=1) #truck.drop(columns='Maintenance Expense','Truck')
X = sm.add_constant(X) # adding a constant

reg_model = sm.OLS(Y,X).fit()
print(reg_model.summary())   

                             OLS Regression Results                            
Dep. Variable:     Maintenance Expense   R-squared:                       0.951
Model:                             OLS   Adj. R-squared:                  0.946
Method:                  Least Squares   F-statistic:                     195.0
Date:                 Tue, 30 Aug 2022   Prob (F-statistic):           7.62e-14
Time:                         08:02:02   Log-Likelihood:                -127.47
No. Observations:                   23   AIC:                             260.9
Df Residuals:                       20   BIC:                             264.3
Df Model:                            2                                         
Covariance Type:             nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const           11.4184     29.191    

/Users/kshyamasagardas/opt/anaconda3/lib/python3.7/site-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


In [ ]:
# from the above data the age of the truck we may discard as the p value is > 0.05 and the condition numer is high.

In [61]:
X = truck.drop(['Maintenance Expense','Truck','Age of Truck'], axis=1) #truck.drop(columns='Maintenance Expense','Truck')
X = sm.add_constant(X) # adding a constant

reg_model = sm.OLS(Y,X).fit()
print(reg_model.summary())   
 

                             OLS Regression Results                            
Dep. Variable:     Maintenance Expense   R-squared:                       0.948
Model:                             OLS   Adj. R-squared:                  0.945
Method:                  Least Squares   F-statistic:                     382.2
Date:                 Tue, 30 Aug 2022   Prob (F-statistic):           5.92e-15
Time:                         08:02:08   Log-Likelihood:                -128.22
No. Observations:                   23   AIC:                             260.4
Df Residuals:                       21   BIC:                             262.7
Df Model:                            1                                         
Covariance Type:             nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const           13.2242     29.398    

/Users/kshyamasagardas/opt/anaconda3/lib/python3.7/site-packages/statsmodels/tsa/tsatools.py:142: FutureWarning: In a future version of pandas all arguments of concat except for the argument 'objs' will be keyword-only
  x = pd.concat(x[::order], 1)


conclusion :
    the equation is y= 13.2242+ 0.08(Mile driven)
     R square for the data set is 0.948 i.e 94.8%
    

In [62]:
#7
fact_overhead = pd.read_csv("MLR_FactoryOverhead.csv")
fact_overhead.sample(5)

Month  Machine Hours  Production Runs  Overhead Cost
30     31           1124               36          88371
6       7           1716               41         117287
5       6           1777               30         107925
16     17           1769               34         110054
9      10           1516               21          88738

In [63]:
fact_overhead.corr()

Month  Machine Hours  Production Runs  Overhead Cost
Month            1.000000       0.037230         0.192256       0.143891
Machine Hours    0.037230       1.000000        -0.229093       0.631885
Production Runs  0.192256      -0.229093         1.000000       0.520544
Overhead Cost    0.143891       0.631885         0.520544       1.000000

does not seems to be any high correlation between the independent variables.
because we need to compare the impact and importance of independent variables we should first standardize the X column in order to get the standardized coifficient. lets work out below

In [66]:
from scipy.stats.mstats import zscore

In [67]:
# Training a linear regression model 
Y = fact_overhead["Overhead Cost"]

X = fact_overhead[['Machine Hours', 'Production Runs']]
std_Y = zscore(Y)
std_X = zscore(X)
print( zscore(X))

[[ 0.42289981 -0.57805615]
 [-0.87811234 -0.81187662]
 [ 0.1729014  -1.04569708]
 [-0.51586974 -1.63024825]
 [ 0.22392148 -0.11041522]
 [ 1.63717782 -0.69496638]
 [ 1.3259553   0.59104618]
 [-2.09749236 -0.81187662]
 [-0.46995167  1.29250757]
 [ 0.30555362 -1.74715848]
 [ 0.85146852  0.12340525]
 [-0.40872756  0.12340525]
 [-0.65872598  1.52632804]
 [-1.41892523  1.64323827]
 [ 0.17800341  0.12340525]
 [ 1.07595689  0.59104618]
 [ 1.59636175 -0.22732545]
 [-1.79647386  0.94177688]
 [-1.32708908  1.17559734]
 [ 1.72391196 -0.81187662]
 [-0.39342154  0.24031548]
 [-0.04138296 -0.46114592]
 [ 0.24943152 -0.46114592]
 [-0.18423919  0.70795641]
 [ 0.09637127 -1.04569708]
 [-1.72504574 -0.22732545]
 [-0.11791308  2.57852013]
 [ 0.67800023 -1.16260731]
 [ 0.6575922  -0.46114592]
 [ 0.18820742 -0.34423569]
 [-1.69443369  0.00649501]
 [ 0.40759378 -0.92878685]
 [ 1.13207898  0.59104618]
 [ 1.36166936 -0.11041522]
 [-0.21995325 -0.69496638]
 [-0.33729945  2.1108792 ]]


In [68]:
std_X = sm.add_constant(std_X) 

# Fit the model on standardized data to get standardized coefficients 
std_reg_model = sm.OLS(std_Y, std_X).fit()
print(std_reg_model.summary())


                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.866
Model:                            OLS   Adj. R-squared:                  0.858
Method:                 Least Squares   F-statistic:                     107.0
Date:                Tue, 30 Aug 2022   Prob (F-statistic):           3.75e-15
Time:                        08:06:41   Log-Likelihood:                -14.846
No. Observations:                  36   AIC:                             35.69
Df Residuals:                      33   BIC:                             40.44
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -4.788e-16      0.064  -7.53e-15      1.0

In [ ]:
# not correlated
# 85%
# machine hr is better predictor

In [ ]:
lets evaluate individually



In [74]:
# Training a linear regression model 
Y = fact_overhead["Overhead Cost"]

X = fact_overhead[['Production Runs']]
std_X = zscore(X)

In [75]:
std_X = sm.add_constant(std_X) 

# Fit the model on standardized data to get standardized coefficients 
std_reg_model = sm.OLS(std_Y, std_X).fit()
print(std_reg_model.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.271
Model:                            OLS   Adj. R-squared:                  0.250
Method:                 Least Squares   F-statistic:                     12.64
Date:                Tue, 30 Aug 2022   Prob (F-statistic):            0.00114
Time:                        08:09:39   Log-Likelihood:                -45.393
No. Observations:                  36   AIC:                             94.79
Df Residuals:                      34   BIC:                             97.95
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const      -4.788e-16      0.146  -3.27e-15      1.0

In [76]:
X = fact_overhead[['Machine Hours',]]
std_X = sm.add_constant(std_X) 
std_X = zscore(X)
# Fit the model on standardized data to get standardized coefficients 
std_reg_model = sm.OLS(std_Y, std_X).fit()
print(std_reg_model.summary())

                                 OLS Regression Results                                
Dep. Variable:                      y   R-squared (uncentered):                   0.399
Model:                            OLS   Adj. R-squared (uncentered):              0.382
Method:                 Least Squares   F-statistic:                              23.26
Date:                Tue, 30 Aug 2022   Prob (F-statistic):                    2.73e-05
Time:                        08:09:52   Log-Likelihood:                         -41.909
No. Observations:                  36   AIC:                                      85.82
Df Residuals:                      35   BIC:                                      87.40
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
                 coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------

 as we saw individually the model performs really poor and its missing the partial effect between the independent variables
    
    hence we will go with the model with both variables and 85% goodness.